src/main/java/TurnUseCases/TradeUseCase/Trade.java

```diff
@@ -0,0 +1,130 @@
+package TurnUseCases.TradeUseCase;
+
+import GameEntities.Player;
+import GameEntities.Tiles.Property;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Contains all the functions related to the trading aspect of this game.
+ */
+
+public class Trade implements  TradeInputBoundary{
+    TemporaryTurnActionPresenter presenter = new TemporaryTurnActionPresenter();
```

Comment:
The `presenter` here should be a object of type `TradeOutputBoundary` that is passed in to the constructor for `Trade`. This way you don't need the `TemporaryTurnActionPresenter` class.

Reply:
Just added a constructor that does this.